kof-operator/webapp/collector/tests/HealthFilter.test.tsx

```diff
@@ -0,0 +1,79 @@
+import { PrometheusTargetsManager } from "../src/models/PrometheusTargetsManager";
+import { Cluster } from "../src/models/Cluster";
+import { State, HealthFilter } from "../src/components/features/HealthSelector";
+import { fakeData } from "./fake_data/fake_response";
+import { describe, it, expect, beforeEach } from "vitest";
+
+describe("Health filter", () => {
+  let prometheusTargetsManager: PrometheusTargetsManager;
+  let clusters: Cluster[];
+
+  beforeEach(() => {
+    prometheusTargetsManager = new PrometheusTargetsManager({
+      clusters: fakeData.clusters,
+    });
+    clusters = prometheusTargetsManager.clusters;
+  });
+
+  it("should return all data unchanged", () => {
+    const filter = HealthFilter([]);
+    const result = filter(clusters);
+
+    expect(result).toEqual(clusters);
+    expect(result).toHaveLength(clusters.length);
+  });
+
+  it("should return clusters with only 'up' targets", () => {
+    const filter = HealthFilter(["up" as State]);
```

Comment:
Please remove "as" from the code, it's generally a bad idea
https://basarat.gitbook.io/typescript/type-system/type-assertion#assertion-considered-harmful
```suggestion
    const filter = HealthFilter(["up"]);
```